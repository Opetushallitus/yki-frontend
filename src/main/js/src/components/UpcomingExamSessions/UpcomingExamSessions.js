import React from 'react';
import PropTypes from 'prop-types';
import { withNamespaces } from 'react-i18next';
import moment from 'moment';

import classes from './UpcomingExamSessions.module.css';
import { DATE_FORMAT, DATE_FORMAT_WITHOUT_YEAR } from '../../common/Constants';
import { languageToString, levelDescription } from '../../util/util';

export const upcomingExamSessions = props => {
  const examSessionRows = props.examSessions.map((e, i) => {
    const registrationOpen = moment().isSameOrAfter(
      moment(e.registration_start_date),
    );
    return (
      <div className={classes.Row} key={i} data-cy={`exam-sessions-table-row-${i}`} onClick={() => props.examSessionSelected(e)}>
        <p>{moment(e.session_date).format(DATE_FORMAT)}</p>
        <p>{languageToString(e.language_code).toLowerCase()}</p>
        <p>{levelDescription(e.level_code).toLowerCase()}</p>
        <p>
          {moment(e.registration_start_date).format(DATE_FORMAT_WITHOUT_YEAR)}
          &ndash;
          {moment(e.registration_end_date).format(DATE_FORMAT)}
        </p>
        <p>
          {registrationOpen ? `${e.participants} / ${e.max_participants}` : '-'}
        </p>
      </div>
    );
  });

  return (
    <div className={classes.ExamSessionList}>
      <h2>{props.t('examSession.upcomingExamSessions')}</h2>
      {props.examSessions.length > 0 ? (
        <div className={classes.Grid} data-cy="exam-sessions-table">
          <h3>{props.t('common.examDate')}</h3>
          <h3>{props.t('common.language')}</h3>
          <h3>{props.t('common.level')}</h3>
          <h3>{props.t('common.registationPeriod')}</h3>
          <h3>{props.t('examSession.participants')}</h3>
          {examSessionRows}
        </div>
      ) : (
        <p>{props.t('examSession.noPlannedSessions')}</p>
      )}
    </div>
  );
};

upcomingExamSessions.propTypes = {
  examSessions: PropTypes.array.isRequired,
  examSessionSelected: PropTypes.func.isRequired
};

export default withNamespaces()(upcomingExamSessions);
