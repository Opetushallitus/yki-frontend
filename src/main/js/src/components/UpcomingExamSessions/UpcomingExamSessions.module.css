.Grid {
  display: grid;
  grid-template-columns: repeat(5, minmax(50px, 1fr));
  grid-row-gap: 8px;
}

.Grid h3 {
  margin: 0;
}

.Grid p {
  margin: 0;
  cursor: pointer;
}

.Heading {
  font-weight: bold;
}

.Row {
  grid-column: 1 / -1;
  display: grid;
  grid-template-columns: repeat(5, minmax(50px, 1fr));
  text-decoration: none;
  color: hsl(0, 0%, 16%);
}

.Row:hover {
  color: hsla(195, 100%, 47%, 1);
  cursor: pointer;
}

@media (max-width: 900px) {
  .DesktopOnly {
    display: none;
  }
}
