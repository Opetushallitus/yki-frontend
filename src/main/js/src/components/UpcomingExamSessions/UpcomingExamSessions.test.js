import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { upcomingExamSessions as UpcomingExamSessions } from './UpcomingExamSessions';

configure({ adapter: new Adapter() });

const examSessions = [
  {
    registration_end_date: '2028-12-15',
    session_date: '2028-12-30',
    participants: 50,
    max_participants: 50,
    registration_start_date: '2018-09-01',
    language_code: 'eng',
    level_code: 'KESKI',
  },
  {
    registration_end_date: '2025-12-15',
    session_date: '2028-05-30',
    participants: 0,
    max_participants: 30,
    registration_start_date: '2028-03-01',
    language_code: 'fin',
    level_code: 'PERUS',
  },
];

jest.mock('i18next', () => ({
  use: () => {
    return { init: () => {} };
  },
  t: k => k,
}));

describe('<UpcomingExamSessions />', () => {
  it('should render 2 exam session rows', () => {
    const wrapper = shallow(
      <UpcomingExamSessions t={key => key} examSessions={examSessions} />,
    );

    expect(wrapper.find('.ExamSessionList').exists()).toBeTruthy();
    expect(wrapper.find('.Row')).toHaveLength(2);
  });
});
