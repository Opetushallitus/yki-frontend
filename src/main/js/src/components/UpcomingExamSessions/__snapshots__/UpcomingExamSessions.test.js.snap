// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpcomingExamSessions /> should render exam session rows 1`] = `
<div
  className="ExamSessionList"
>
  <h2>
    examSession.upcomingExamSessions
  </h2>
  <div
    className="Grid"
    data-cy="exam-sessions-table"
  >
    <h3>
      common.examDate
    </h3>
    <h3>
      common.language
    </h3>
    <h3>
      common.level
    </h3>
    <h3>
      common.registationPeriod
    </h3>
    <h3>
      examSession.participants
    </h3>
    <div
      className="Row"
      data-cy="exam-sessions-table-row-0"
      key="0"
      onClick={[Function]}
    >
      <p>
        30.12.2028
      </p>
      <p>
        englanti
      </p>
      <p>
        common.level.middle
      </p>
      <p>
        1.9.
        –
        15.12.2028
      </p>
      <p>
        50 / 50
      </p>
    </div>
    <div
      className="Row"
      data-cy="exam-sessions-table-row-1"
      key="1"
      onClick={[Function]}
    >
      <p>
        30.5.2028
      </p>
      <p>
        suomi
      </p>
      <p>
        common.level.basic
      </p>
      <p>
        1.3.
        –
        15.12.2025
      </p>
      <p>
        -
      </p>
    </div>
  </div>
</div>
`;
